src/Products.jl

```diff
@@ -207,6 +207,86 @@ function locate(lp::LibraryProduct, prefix::Prefix; platform::Platform = platfor
     return nothing
 end
 
+"""
+A `FrameworkProduct` is a  [`Product`](@ref) that encapsulates a macOS Framework.
+It behaves mostly as a [`LibraryProduct`](@ref) and on non-macOS platforms falls back
+to the behavior of `LibraryProduct`, so there is no need to define the products
+differently on non-macOS platforms.
```

Comment:
To be honest, I'm not overly happy about this in principle.  However, I guess we aren't going to use this product very often, and on the other hand this does make things much easier, so in the end I'm happy enough